api/cds.proto

```diff
@@ -0,0 +1,233 @@
+syntax = "proto3";
+
+import "api/address.proto";
+import "api/node.proto";
+import "api/ssl_context.proto";
+
+import "google/protobuf/wrappers.proto";
+
+service ClusterDiscoveryService {
+  // Return list of all clusters, this proxy will load balance to.
+  rpc StreamClusters(ClusterDiscoveryRequest)
+      returns (stream ClusterDiscoveryResponse) {
+  }
+}
+
+message ClusterDiscoveryRequest {
+  Node node = 1;
+}
+
+message ClusterDiscoveryResponse {
+  repeated Cluster cluster = 1;
+}
+
+message Duration {
+  google.protobuf.UInt32Value milliseconds = 1;
+}
+
+message HealthCheck {
```

Comment:
Shouldn't this be in separate file? HealthCheck will be used by both CDS and HDS.